components/EmotionForm.js

```diff
@@ -130,7 +143,8 @@ export default function EmotionForm({
             name="dateTime"
             type="datetime-local"
             defaultValue={
-              defaultValue?.dateTime || currentDate + "T" + currentTime
+              selectedDate + "T" + currentTime ||
+              currentDate + "T" + currentTime
```

Comment:
selectedDate is set to currentDate in the useState callback. You might be able to remove this second statement after the `||` operator.